File: bp-agent/src-go/llm/codex_adapter.go
package llm

import (
	"context"
)

// CodexConfig configures Codex adapter.
type CodexConfig struct {
	APIKeys         []string
	AuthFiles       []string
	Model           string
	ReasoningEffort string
}

// CodexAdapter is a placeholder implementation.
type CodexAdapter struct {
	cfg     CodexConfig
	rotator *Rotator
}

// NewCodexAdapter creates a new adapter.
func NewCodexAdapter(cfg CodexConfig) *CodexAdapter {
	return &CodexAdapter{cfg: cfg, rotator: NewRotator(cfg.APIKeys)}
}

// Complete calls the Codex API.
func (a *CodexAdapter) Complete(ctx context.Context, req CompletionRequest) (LLMResponse, error) {
	_ = ctx
	return LLMResponse{}, ProviderError{Provider: "codex", Message: "adapter not implemented"}
}

File: bp-agent/src-rs/llm/codex_adapter.rs
use super::rotation::Rotator;
use super::types::{CompletionRequest, LLMResponse, ProviderAdapter, ProviderError};

pub struct CodexConfig {
    pub api_keys: Vec<String>,
    pub auth_files: Vec<String>,
    pub model: String,
    pub reasoning_effort: Option<String>,
}

pub struct CodexAuth {
    pub auth_file: String,
}

pub struct CodexAdapter {
    cfg: CodexConfig,
    rotator: Rotator,
}

impl CodexAdapter {
    pub fn new(cfg: CodexConfig) -> Self {
        Self {
            rotator: Rotator::new(cfg.api_keys.clone()),
            cfg,
        }
    }
}

impl ProviderAdapter for CodexAdapter {
    fn complete(&self, _request: CompletionRequest) -> Result<LLMResponse, ProviderError> {
        let _ = &self.rotator;
        Err(ProviderError::new("not_implemented", "codex adapter not implemented", false))
    }
}
